src/foam/nanos/auth/ServiceProvider.js

```diff
@@ -84,17 +85,9 @@ foam.CLASS({
           // find the ucj and update its status to granted, or create a ucj none found
           
           UserCapabilityJunction ucj;
+          Subject subject = new Subject.Builder(x).setUser(user).build();
```

Comment:
FYI: I created convenience constructor  new Subject(x, user)  for just this type of thing.

Reply:
thanks, updated